### ISSUE TITLE
Extended information in accesslog 

### ISSUE THREAD

**Issue by npateriya** [containous/traefik#155]
Currently access include only relative url information. 
In cases where traefik is configured to serve multiple microservices in backend just relative url does not  provide sufficient information to relate access log to specific microservice. Example these 2 log entries are from 2 different  microservices but no way to identify from log data.  
```
173.39.245.211 - - [06/Jan/2016:23:56:15 +0000] "GET /events/count HTTP/1.0" 200 12 "" "Go 1.1 package http"
173.39.245.197 - - [06/Jan/2016:23:56:14 +0000] "GET /projects HTTP/1.0" 200 12671 "" "Go 1.1 package http"
```

Quick look at code it seems currently gorilla combinedloghandler  is used, combinedloghandler content is good for single microservice access log but for proxy additional information will be helpful. 

Log handler capturing equivalent of http://www.haproxy.org/download/1.4/doc/configuration.txt (8.2.3. HTTP log format) that also includes front_end, backend_end and Total time. 

Is implementing a custom log handler part of traefik that also does access logging including field relevant to proxy  sounds reasonable? We can also contribute once high level design is agreed.

**Comment by emilevauge** [comment]
Hi @npateriya, this sounds totally reasonable :)
It would be great if could contribute on that point.
What fields do you need?
* front_end
* backend_end
* Total time
That's all ?

**Comment by KevinRandolph** [comment]
Hi @emilevauge, I'm working with @npateriya to implement the additional logging information.

I cloned github.com/gorilla/handlers/handlers.go into traefik/middlewares/logger.go, then modified the logging code.  That works, but @npateriya expressed concern about the inclusion of code from gorilla.  The gorilla/handlers/handlers.go code is governed a BSD-style license that permits redistribution of the modified source code as long as their copyright and disclaimers are preserved.

What are your feelings on this?  Is it OK to use the gorilla/handlers/handlers.go code in traefik/middlewares/logger.go, or should I attempt to log all of the information myself, adhering to the 8.2.3. HTTP log format specification?

On a related topic, I'm looking for the appropriate back_end URL to log.  What would be the best way to identify the back_end URL that serviced the request?

Thanks,
Kevin

**Comment by emilevauge** [comment]
Fixed by #215
